### ISSUE TITLE
dotnet --info shows runtimes and sdks twice

### ISSUE THREAD

**Issue by fubar-coder** [dotnet/cli#10966]
Reopened issue (was: #10479)

## Steps to reproduce

1. Ensure that the "Path" environment variable contains "c:\program files\dotnet" (with the same casing)
2. dotnet --info

## Expected  behavior

Show every SDK and runtime once.

## Actual behavior

Shows every SDK and runtime twice.

## What I found out

When I change the "Path" environment variable to use "C:\Program Files\dotnet", then everything works as expected. It seems that there is a case-sensitive `Distinct` list of SDK paths even though it should be case-insensitive on Windows.

## Environment data
`dotnet --info` output:

```text
.NET Core SDK (gemäß "global.json"):
 Version:   2.2.101
 Commit:    236713b0b7

Laufzeitumgebung:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   c:\program files\dotnet\sdk\2.2.101\

Host (useful for support):
  Version: 2.2.0
  Commit:  1249f08fed

.NET Core SDKs installed:
  1.0.4 [C:\Program Files\dotnet\sdk]
  1.0.4 [c:\program files\dotnet\sdk]
  2.1.201 [C:\Program Files\dotnet\sdk]
  2.1.201 [c:\program files\dotnet\sdk]
  2.1.202 [C:\Program Files\dotnet\sdk]
  2.1.202 [c:\program files\dotnet\sdk]
  2.1.301 [C:\Program Files\dotnet\sdk]
  2.1.301 [c:\program files\dotnet\sdk]
  2.1.502 [C:\Program Files\dotnet\sdk]
  2.1.502 [c:\program files\dotnet\sdk]
  2.2.101 [C:\Program Files\dotnet\sdk]
  2.2.101 [c:\program files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.1 [c:\program files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.6 [c:\program files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.0 [c:\program files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.1 [c:\program files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.6 [c:\program files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.0 [c:\program files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.5 [c:\program files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.2 [c:\program files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.7 [c:\program files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.9 [c:\program files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.1 [c:\program files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.6 [c:\program files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.0 [c:\program files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by livarcocc** [comment]
Do you have an active repro of this? I could never reproduce this issue.

**Comment by livarcocc** [comment]
cc @peterhuene who also tried it.

**Comment by peterhuene** [comment]
I think in the past I wasn't aware of the case differences in `PATH` as part the repo.  I'll see if I can reproduce this again with this information.

**Comment by livarcocc** [comment]
@peterhuene This print out actually comes from the host. The case-sensitive comparison makes sense there. If you confirm the repro, I would suggest moving this issue to core-setup.

**Comment by fubar-coder** [comment]
I didn't recognize the casing differences too - until today.

**Comment by peterhuene** [comment]
It looks like this was fixed for the [3.0 muxer](https://github.com/dotnet/core-setup/blob/1d95ece515edbbad0ef8f8980424d3e3bce49cb2/src/corehost/common/utils.cpp#L360).

Basically the bug was that if the `dotnet` found on the PATH was in location that was a case-insensitive match with the "global" locations on Windows (e.g. `C:\Program Files\dotnet`), there would be duplicate SDK and frameworks output because multiple items would be put into this list of locations to probe.

The above line now only inserts into the list if it isn't a case-insensitive match with `dotnet`'s location.

As I believe this has been fixed in 3.0, I am closing this issue.  Please reactivate if you're able to reproduce this with a 3.0 SDK.

Thanks for reporting this @fubar-coder!
